---
title: "WM_selection"
author: "Nieto_Lopez"
date: "8/29/2018"
output:
  html_document: default
  pdf_document: default
---
```{r setup, include=FALSE}
#Image: ![](Documents/to/white_mold.png)
library(tidyverse)
library(dplyr)
library(ggplot2)
library(knitr)
library(maps)
library(mapdata)
library(geosphere)
library(sf)
library(lwgeom)
knitr::opts_chunk$set(echo = TRUE)
# graph names of the body means "number isolates/field" 
# graph names of powdery mildews means "Year"

```
###Reading data 
```{r}
###
WM <- read.csv("data/WM-fixingproductnamesID2416.csv", stringsAsFactors = TRUE)# Making the columns as factors
#filtering 
WM <-  WM %>% 
   filter(!Collection_ID== "429", !Collection_ID== "433", !Collection_ID== "442", !Collection_ID== "1048", !Collection_ID== "1049", !Collection_ID== "2217", !Collection_ID== "2219", !Collection_ID== "2226", !Collection_ID== "2245", !Collection_ID== "2307", !Collection_ID== "2327", !Collection_ID== "2352", !Collection_ID== "2366", !Collection_ID== "2367", !Collection_ID== "2380", !Collection_ID== "2392"  )
#WM<- WM[-c(1055,419,423, 432),]# Taking out this row that is in blank
WM$long <- as.numeric(as.character(WM$long))# Transforming long variables to numeric
#WM<- WM[-1293, ]
WM$long <- (WM$long) * -1 # Transforming the long variable to the Hemispher West side


#WM$Collection_ID=="2215"
WM$Year <- as.factor(WM$Year)# Changing the level of year to factor
#the following markas an error that is why it was uncripted  
WM$Collection_ID <- as.character(as.numeric(WM$Collection_ID))# The goal is to change  the level of Collection to numeric, but for some reason, it should be by step by step. Changing the level of year to numeric in first approach 
 # WM$Collection_ID <- as.character(WM$Collection_ID)#Changing the level of year to character as second approach
  levels(WM$Origin )[levels(WM$Origin ) == "trial"] <- "Fungicide field trials"#Changing the name to be more specific
  levels(WM$Origin  )[levels(WM$Origin ) == "Survey"] <- "Farmer fields"#Changing the name to be more specific
  levels(WM$DNA.Extraction )[levels(WM$DNA.Extraction ) == ""] <- "No DNA extraction"#Changing the name to be more specific
  levels(WM$DNA.Extraction )[levels(WM$DNA.Extraction ) == "No"] <- "No DNA extraction"#Changing the name to be more specific
  levels(WM$DNA.Extraction )[levels(WM$DNA.Extraction ) == "Yes"] <- "Yes DNA extraction"#Changing the names to be more specific
usa <- map_data("usa")# USA map 
states <- map_data("state")# USA map by states
counties <- map_data("county")# USA map by counties
bra <- map_data("worldHires", "Brazil")
mexico <- map_data("worldHires", "Mexico")

mid_west_county <-
  subset(
    counties,
    region == "nebraska" |
      region == "wisconsin" | region == "michigan" | region == "iowa"
  )

```
Before produce Knit run line 263 twice    
## Backgroundd   
Fungicide applications are essential for maintaining healthy, reliable, and high-quality agricultural products. Resistance to fungicides is an increasing threat to this effectiveness and has already been reported for 203 plant pathogenic fungi (Fungicide Resistance Action Committee 2013). Soybean is the second most important crop in the U.S. and contributes significantly to the nation’s economy with annual sales of $38.7 billion, accounting for 10 percent of total U.S. agriculture sales (Census of Agriculture 2012). In the temperate North Central soybean production areas of the United States, the white mold disease can be a significant yield limiting disease, with reported losses more than 10 million bushels per year (Peltier et al. 2012). This yield loss is significant and makes S. sclerotiorum one of the most important pathogens of soybean in the North Central U.S. In Mexico, dry common bean (_Phaseolus vulgaris_) is the second most important crop and is also significantly affected by _S._ _sclerotiorum_. For example, in the State of Sinaloa, _S._ _sclerotiorum_ causes up to 75% yield loss in dry bean and is considered the most important fungal pathogen (Apodaca et al. 2011).
Lack of adequate levels of host resistance means that fungicides are the primary method of disease control (Bardin and Huang 2001). A major problem is that by the time fungicide resistance is detected by growers, it may be too late to prevent spread of fungicide resistance. Identifying changes in fungicide sensitivity requires knowledge of existing baseline sensitivity to each class of fungicides. Acquisition of fungicide resistance within a completely susceptible population begins with emergence and is followed by selection and adjustment phases. Few studies have addressed the emergence phase and there is a debate about the role of fungicide dose (Van Den Bosch et al. 2011). In the literature, authors speculate that rate of resistance emergence may be directly related to pathogen exposure to low-doses of fungicides (Gressel 2011), which may increase with fewer fungicide applications, such as is recommended in an Integrate Pest Management (IPM) method (Beckerman et al. 2015). This is relevant to soybean production because low-doses of fungicides are likely to occur in the field, whether due to intentional reduction of application rates or unintentionally because of incomplete spray coverage. Other factors related to risk of resistance emergence in fungal plant pathogens is the mechanism of reproduction of the fungus (sexual out-crossing, sexual selfing or asexual reproduction) and population structure of the pathogen. Knowledge of the population structure and spread of _S._ _sclerotiorum_ on soybean and dry bean is needed.  

## THE FIRST PART OF DISSERTATION, Survey fungicide sensitivity of populationss of S. sclerotiorum from soybean
#Backgroundddd
Knowledge of baseline sensitivity of a pathogen population to fungicides is important for detection of resistance and for achieving effective disease management strategies (Brent and Holloman 2007). QoI, DMI, MBC and SDHI fungicides are used to control _S._ _sclerotiorum_ in an IPM approach in soybean and dry bean in the U.S. DMI fungicides are considered to medium risk of resistance emergence, whereas QoI, SDHI and MBC are high risk (Fungicide Resistance Action Committee 2016). MBC and dicarboximide fungicides are used for white mold in Brazil, wherein resistance to MBC fungicide, Thiophanate methyl, has been reported (Lehner et al. 2015). Fungicide resistance has also been reported in China to the MBC, Carbendazim and dicarboximide, Dimethachlon (Ma et al. 2009). Field application of fungicides in the United States on soybean show control of S. sclerotiorum is low to fair for QoI, DMI and MBC fungicides, and good for SDHI fungicides (The North Central Regional Committee on Soybean Diseases 2015). However, it is unknown whether efficacy is related to reduce sensitivity within populations under study. Therefore, the __hypothesis of this objective is that there are isolates of _S. sclerotiorum_ with low sensitivity to the most commonly used fungicides in the North Central United States.__  

__Results will be used to determine whether some isolates are resistant to a given fungicide. It is expected that resistance to fungicides with single-site modes (β -tubulin assembly in mitosis, C14 demethylase in sterol biosynthesis, complex III cytochrome bc1 at Qo site and complex II succinate-dehydro-genase) of action will be more common than for multi-site fungicides__



   
## Objective   
 1. Select hierarchieally isolates by space according 


### Setting some objects previuosly work with
```{r}
#Aready susbsitited the formID of baselines isolates for the Collection_ID  
tested <-# #Fungicide pre-tested isolates from USA midwest including baselines and  Mexican isolates
  c(
    "1",
    "118",
    "123",
    "12B",
    "129",
    "20",
    "21",
    "449",
    "461",
    "467",
    "475",
    "558",
    "564",
    "568",
    "581",
    "645",
    "800",
    "667",
    "74-SS-1",
    "8",
    "87",
    "2390" ,
    "2386",
    "2385",
    "2098",
    "2099",
    "2100",
    "2139",
    "2140",
    "2143",
    "2220",
    "2222",
    "2223",
    "2362",
    "2320",
    "2388",
    "1025",
    "1026",
    "1027",
    "1029",
    "1032",
    "1033",
    "1870",#mexican isolates
    "1872",#mexican isolates
    "1884",#mexican isolates
    "1885",#mexican isolates
    "2453", #54c
    "2528",#65B
    "2449",#53b
    "2505",# 71B
    "2518", #64D
    "2536", #53b
  "2559",  #"60A",
    "698",
    "699",
    "710",
    "711",
    "724",
    "725",
    "731",
    "732",
    "738",
    "739",
    "746",
    "751",
    "755",
    "756",
    "757",
    "764",
    "765",
    "771",
    "772",
    "786",
    "787",
    "811",
    "812",
    "813",
    "814",
    "817",
    "818",
    "851",
    "852",
    "853",
    "855",
    "858",
    "859",
    "860",
    "861",
    "862",
    "867",
    "870",
    "871",
    "877",
    "878",
    "884",
    "885",
    "891",
    "892",
    "896",
    "897",
    "901",
    "902",
    "905",
    "906",
    "908",
    "909",
    "911",
    "912",
    "914",
    "274",
    "307",
    "504",
    "505",
    "2384",
    "2385",
    "2386",
    "2388",
    "2407",
    "2408",
    "2383",
    "1058",
    "1081",
    "1087",
    "1109",
    "1127",
    "1128",
    "1134",
    "1135",
    "1139",
    "1175",
    "1026",
    "1027",
    "1029",
    "1328",
    "1329",
    "1330",
    "1331",
    "1332",
    "1340",
    "1345",
    "1365",
    "1366",
    "1392",
    "1327",
    "1502",
    "1582",
    "1620",
    "1622",
    "1541",
    "1671",
    "1672",
    "1691",
    "1692",
    "1712",
    "1713",
    "1721",
    "1722",
    "1731",
    "1732",
    "1791"
    # "1",  #since here repeated
    #  "118",
    #  "123",
    #  "20",
    #  "74-SS-2",
    #  "8",
    #  "2385",
    #  "2140",
    #  "2220",
    #  "2222"
  )
mexican_isolates <- c("242","248","248a","248b","615":"632", "779","1857":"1940" )#All isolates from Mexico
brazilian_isolates <- c("400","400a","400b","402","972",
"973", 
"973A",
"973B",
"973C",
"973D",
"973E",
"974",
"974A",
"974B",
"974C",
"974D",
"974E",
"975",
"975A",
"975B",
"975C",
"975D",
"975E",
"976",
"976A",
"976B",
"976C",
"976D",
"976E",
"977",
"977A",
"977B",
"977C",
"977D",
"977E",
"978",
"978A", 
"978B", 
"978C",
"978D",
"978E",
"979",
"979A",
"979B",
"979C",
"979D",
"979E", "1010":"1022","1227":"1254","2417":"2569" )#All isolates from Brazil
baseline_isolates <- c("1","118", "123", "12B", "129", "20", "21", "449", "61", "467", "475", "558", "564", "568", "581", "645", "800", "667", "74-SS-1", "8", "87")
white_mold_nursery <- c( "449", "461", "467", "475", "558", "564", "568", "581", "645", "800", "667")

survey_before_fungicides_lunched <- c("1","118", "123", "12B", "129", "20", "21",  "74-SS-1", "8", "87")
 tested[duplicated(tested)]
 tested <- tested[!duplicated(tested)]
```

##White Mold selection from USA midwest  
###Data wrangling
```{r}
  wm1 <-
  WM %>% select(
    Collection_ID,
    Host, 
    State,
    Field,
    Year,
    Origin,
    long,
    lat,
    hyphal_tip..ht.,
    DNA.Extraction,
    County,
    Country,
    Serial_agar_dilution_for_creating_the_model_..N.20.,
    Form.ID 
      )# Selecting 16 the most important columns so far

wm1_ <-  wm1 %>%
  mutate(Fungicide_status = ifelse(Collection_ID  %in% tested,
                                   "tested", "no_tested")) %>%# Creating a new column based wheather or not funcides pretested
 # filter(!Collection_ID %in% mexican_isolates) %>%#Taking out mexican isolates
  group_by(Field) %>% mutate (num_isolates_per_field = ifelse(n() > 2, "≥3isolates/Field", "<3isolates/Field")) %>%#Creating a new column if the number of isolates/field is major than 2
  ungroup()

wm1_$num_isolates_per_field <- as.factor(wm1_$num_isolates_per_field)#Converting as factor the levels of the new columns

levels(wm1_$Origin) = c("", "Farmer fields", "Fungicide field trials", "Baseline isolates")#, "Farmer fields before fungicides lunched")


wm1_[(which(wm1_$Collection_ID %in% baseline_isolates)), 6] <- "Baseline isolates"

#wm1_[(which(wm1_$Collection_ID %in% survey_before_fungicides_lunched)), 6] <- "before fungicides lunched"

wm1__ <- wm1_ %>% select(Collection_ID, State, County, Country, Origin, Year, Fungicide_status, Host) %>% filter(Fungicide_status == "tested", Host== "drybeans") 
q <- wm1__ %>%  group_by(State) %>% summarize(n=n())
ccc <-  wm1__ %>%  group_by(State, Origin) %>% summarize(n=n())
 
wm1___ <- wm1_ %>% select(Collection_ID, State, County, Country, Origin, Year, Fungicide_status, Host) %>% filter(Fungicide_status == "tested", Host== "soybeans") 
qq <- wm1___ %>%  group_by(State, Origin) %>% summarize(n=n())
 wm1__ %>%  group_by(State, Origin) %>% summarize(n=n())

```
###Cleaning and making some arrangemments from the inventory so far, run twice
```{r, warning= FALSE}
# wm1_ <- wm1_[-970, ]# Taking out this column that is in blank
# wm1_$long <- as.numeric(as.character(wm1_$long))# Transforming long variables to numeric
# wm1_$long <- (wm1_$long) * -1 # Transforming the long variable to the Hemispher West side

```
##White Mold selection from USA midwest  and MExico 
###Data wrangling
```{r}
#install.packages("geosphere")
#library(geosphere)
# reading data from fungicide senitivity apporach
combiningfungicides_outliersEC50DC <- read.csv("data/combiningfungicides_outliersEC50DC.csv")
combiningfungicides_outliersEC50DC <- combiningfungicides_outliersEC50DC %>% select(-X)
combiningfungicides_outliersEC50DC$ID <- as.character(combiningfungicides_outliersEC50DC$ID)
combiningfungicides_outliersEC50DC$ID[2]="2220"#these the ID of the H-01

# 

 isolateshighEC50 <- wm1 %>%  
   mutate(field_selection = ifelse(Collection_ID  %in%  #using the isolates with highh EC50
                                     combiningfungicides_outliersEC50DC$ID,
                                   "field_selected", "field_no_selected")) %>% filter(field_selection =="field_selected")
isolateshighEC50

  # c(po  lan  na  han mon  ini ley 75  Hi)
wm2_ <-  wm1 %>%
  mutate(Fungicide_status = ifelse(Collection_ID  %in% tested,
                                   "tested", "no_tested")) %>%# Creating a new column based wheather or not funcides pretested
  #filter(Collection_ID %in% mexican_isolates) %>%#Conisdering mexican isolates
  # group_by(Field) %>% mutate (num_isolates_per_field = ifelse(n() > 2, "≥3isolates/Field", "<3isolates/Field")) %>%#Creating a new column if the number of isolates/field is major than 2
  ungroup()

#taking 9 samples for each field selected and no taking those isolates already done in the pre test by  the combiningfungicides_outliersEC50DC object
selection_nice <- wm2_  %>%
  ungroup() %>%
  filter(
  Field == "po" |
  Field == "lan" |
  Field == "na" |
  Field == "han" |
  Field == "mon" |
  Field == "ini" | Field == "ley" | Field == "75" | Field == "hi"
  ) %>%
  filter(
  !Collection_ID == "1029",
  !Collection_ID == "1087",
  !Collection_ID == "1135",
  !Collection_ID == "1345",
  !Collection_ID == "1541",
  !Collection_ID == "1692",
  !Collection_ID == "1713",
  !Collection_ID == "1732",
  !Collection_ID == "1872",
  !Collection_ID == "1884",
  !Collection_ID == "2131",
  !Collection_ID == "2020"
  ) %>% group_by(Field) %>%
  sample_n(9, replace = F)
  # takingout those fields selected  
selection_chingon <- wm2_  %>%
  ungroup() %>%
  filter(
  !Field == "po" ,
  !Field == "lan" ,
  !Field == "na" ,
  !Field == "han" ,
  !Field == "mon" ,
  !Field == "ini" , 
  !  Field == "ley" , 
  !  Field == "75" , 
   !Field == "hi"
  ) 
#selecting the unique fields
#write_csv(dat_long_lat, "fields_WMNCSRP.csv")
#[-c(,6,12,13,65,67:70,72:78,80:87, 89:97,100:108,110),]
 #dat_long_lat <-  dat_long_lat[-598,]

fields_WMNCSRP <-selection_chingon %>% 
  select(State, County, Field, long, lat) %>% 
  distinct()
a <- fields_WMNCSRP[-c(3,6,12,13,38, 65,67:70,72:78,80:87, 89:97,100:108,110),]
#[-c(3,6,12,13,38, 65,67:70,72:78,80:87, 89:97,100:108,110),]
#a<- read.csv("fields_WMNCSRP.csv")

#a <- a[-34,]# because there are two "48" in Boone county
# COnvert to sf object
b<- st_as_sf(a, coords = c("long", "lat"))
#is Set the projection as ESPG 4326 (long_lat)

st_crs(b) <- 4326

# Apply the st_distance function

dist_m <- st_distance(b)# euclidian distance
# Combine with df
dist_matrix <- b %>% 
  mutate("flo" = as.numeric(dist_m[, 1]), 
                       "ho" = as.numeric(dist_m[, 2]),
                       "an" = as.numeric(dist_m[, 3]),
                       "dod" = as.numeric(dist_m[, 4]),
                       "va" = as.numeric(dist_m[, 5]),
                       "sa" = as.numeric(dist_m[, 6]),
                       "cu" = as.numeric(dist_m[, 7]),
                       "ant" = as.numeric(dist_m[, 8]),
                       "hold" = as.numeric(dist_m[, 9]),
                       "arr" = as.numeric(dist_m[, 10]),
                       "pre" = as.numeric(dist_m[, 11]),
                       "ru" = as.numeric(dist_m[, 12]),
                       "ca" = as.numeric(dist_m[, 13]),
                       "st" = as.numeric(dist_m[, 14]),
                       "6" = as.numeric(dist_m[, 15]),
                       "8" = as.numeric(dist_m[, 16]),
                       "9" = as.numeric(dist_m[, 17]),
                       "10" = as.numeric(dist_m[, 18]),
                       "11" = as.numeric(dist_m[, 19]),
                       "18" = as.numeric(dist_m[, 20]),
                       "19" = as.numeric(dist_m[, 21]),
                       "20" = as.numeric(dist_m[, 22]),
                       "21" = as.numeric(dist_m[, 23]),
                       "22" = as.numeric(dist_m[, 24]),
                       "23" = as.numeric(dist_m[, 25]),
                       "24" = as.numeric(dist_m[, 26]),
                       "25" = as.numeric(dist_m[, 27]),
                       "34" = as.numeric(dist_m[, 28]),
                       "35" = as.numeric(dist_m[, 29]),
                       "45" = as.numeric(dist_m[, 30]),
                       "46" = as.numeric(dist_m[, 31]),
                       "47" = as.numeric(dist_m[, 32]),
                       "48" = as.numeric(dist_m[, 33]),
                       "60" = as.numeric(dist_m[, 34]),
                       "62" = as.numeric(dist_m[, 35]),
                       "66" = as.numeric(dist_m[, 36]),
                       "67" = as.numeric(dist_m[, 37]),
                       "68" = as.numeric(dist_m[, 38]),
                       "69" = as.numeric(dist_m[, 39]),
                       "78" = as.numeric(dist_m[, 40]),
                       "97" = as.numeric(dist_m[, 41]),
                       "98" = as.numeric(dist_m[, 42]),
                       "99" = as.numeric(dist_m[, 43]),
                       "100" = as.numeric(dist_m[, 44]),
                       "101" = as.numeric(dist_m[, 45]),
                       "102" = as.numeric(dist_m[, 46]),
                       "103" = as.numeric(dist_m[, 47]),
                       "104" = as.numeric(dist_m[, 48]),
                       "10" = as.numeric(dist_m[, 49]),
                       "Al" = as.numeric(dist_m[, 50]),
                       "Hi" = as.numeric(dist_m[, 51]),
                       "ln" = as.numeric(dist_m[, 52]),
                       "mont" = as.numeric(dist_m[, 53]),
                       "Sand" = as.numeric(dist_m[, 54]),
                       "Sani" = as.numeric(dist_m[, 55]),
                       "st" = as.numeric(dist_m[, 56]),
                       "howard" = as.numeric(dist_m[, 57]),
                       "chi" = as.numeric(dist_m[, 58]),
                       "floyd" = as.numeric(dist_m[, 59]),
                       "sto" = as.numeric(dist_m[, 60]),
                       "sh" = as.numeric(dist_m[, 61]),
                       "be" = as.numeric(dist_m[, 62]),
                       "ta" = as.numeric(dist_m[, 63]),
                       "wau" = as.numeric(dist_m[, 64]),
                       "la" = as.numeric(dist_m[, 65]),
                       "co" = as.numeric(dist_m[, 66]),
                       "cern" = as.numeric(dist_m[, 67]),
                       "cer" = as.numeric(dist_m[, 68]))
# Replace 0 with NA
#selecting the unique fields
#write_csv(dat_long_lat, "fields_WMNCSRP.csv")
#[-c(2,6,12,13,65,67:70,72:78,80:87, 89:97,100:108,110),]
 #dat_long_lat <-  dat_long_lat[-598,]


dist_matrix[dist_matrix == 0] <- NA              
# object to hierarchially cluster 
hc <- hclust(as.dist(dist_m), method="complete")
# 50 km of distance for each one
d <- 50000

dist_matrix$cluster <- cutree(hc, h=d)
#plot(dist_matrix$cluster)


#### EXAMPLE FOR PART OF GENOTYPING, no filterrin from datamatrix

sclerotiourum <-  wm2_%>% 
  select(State, County, Field, long, lat) %>% 
 distinct ()
  
 sclerotiourum<- sclerotiourum[-c(5,8,11:28, 35:36, 63,91, 93:96,98:104, 106:113, 115:123, 126:134,136),]
# COnvert to sf object

sf_sclerotiourum<- st_as_sf(sclerotiourum, coords = c("long", "lat"))
# Set the projection as ESPG 4326 (long_lat)

st_crs(sf_sclerotiourum) <- 4326
# Apply the st_distance function

dist_m_sclerotiourum <- st_distance(sf_sclerotiourum)
# Combine with df
sclerotiourum_dist_matrix <- sf_sclerotiourum %>%
  mutate(
  "po" = as.numeric(dist_m_sclerotiourum[, 1]),
  "flo" = as.numeric(dist_m_sclerotiourum[, 2]),
  "lan" = as.numeric(dist_m_sclerotiourum[, 3]),
  "ho" = as.numeric(dist_m_sclerotiourum[, 4]),
  "an" = as.numeric(dist_m_sclerotiourum[, 5]),
  "dod" = as.numeric(dist_m_sclerotiourum[, 6]),
  "va" = as.numeric(dist_m_sclerotiourum[, 7]),
  "na" = as.numeric(dist_m_sclerotiourum[, 8]),
  "sa" = as.numeric(dist_m_sclerotiourum[, 9]),
  "han" = as.numeric(dist_m_sclerotiourum[, 10]),
  "cu" = as.numeric(dist_m_sclerotiourum[, 11]),
  "mon" = as.numeric(dist_m_sclerotiourum[, 12]),
  "ant" = as.numeric(dist_m_sclerotiourum[, 13]),
  "holt" = as.numeric(dist_m_sclerotiourum[, 14]),
  "arr" = as.numeric(dist_m_sclerotiourum[, 15]),
  "ini" = as.numeric(dist_m_sclerotiourum[, 16]),
  "ley" = as.numeric(dist_m_sclerotiourum[, 17]),
  "pre" = as.numeric(dist_m_sclerotiourum[, 18]),
  "ru" = as.numeric(dist_m_sclerotiourum[, 19]),
  "ca" = as.numeric(dist_m_sclerotiourum[, 20]),
  "st" = as.numeric(dist_m_sclerotiourum[, 21]),
  "6" = as.numeric(dist_m_sclerotiourum[, 22]),
  "8" = as.numeric(dist_m_sclerotiourum[, 23]),
  "9" = as.numeric(dist_m_sclerotiourum[, 24]),
  "10" = as.numeric(dist_m_sclerotiourum[, 25]),
  "11" = as.numeric(dist_m_sclerotiourum[, 26]),
  "18" = as.numeric(dist_m_sclerotiourum[, 27]),
  "19" = as.numeric(dist_m_sclerotiourum[, 28]),
  "20" = as.numeric(dist_m_sclerotiourum[, 29]),
  "21" = as.numeric(dist_m_sclerotiourum[, 30]),
  "22" = as.numeric(dist_m_sclerotiourum[, 31]),
  "23" = as.numeric(dist_m_sclerotiourum[, 32]),
  "24" = as.numeric(dist_m_sclerotiourum[, 33]),
  "25" = as.numeric(dist_m_sclerotiourum[, 34]),
  "34" = as.numeric(dist_m_sclerotiourum[, 35]),
  "35" = as.numeric(dist_m_sclerotiourum[, 36]),
  "45" = as.numeric(dist_m_sclerotiourum[, 37]),
  "46" = as.numeric(dist_m_sclerotiourum[, 38]),
  "47" = as.numeric(dist_m_sclerotiourum[, 39]),
  "48" = as.numeric(dist_m_sclerotiourum[, 40]),
  "60" = as.numeric(dist_m_sclerotiourum[, 41]),
  "62" = as.numeric(dist_m_sclerotiourum[, 42]),
  "66" = as.numeric(dist_m_sclerotiourum[, 43]),
  "67" = as.numeric(dist_m_sclerotiourum[, 44]),
  "68" = as.numeric(dist_m_sclerotiourum[, 45]),
  "69" = as.numeric(dist_m_sclerotiourum[, 46]),
  "75" = as.numeric(dist_m_sclerotiourum[, 47]),
  "78" = as.numeric(dist_m_sclerotiourum[, 48]),
  "97" = as.numeric(dist_m_sclerotiourum[, 49]),
  "98" = as.numeric(dist_m_sclerotiourum[, 50]),
  "99" = as.numeric(dist_m_sclerotiourum[, 51]),
  "100" = as.numeric(dist_m_sclerotiourum[, 52]),
  "101" = as.numeric(dist_m_sclerotiourum[, 53]),
  "102" = as.numeric(dist_m_sclerotiourum[, 54]),
  "103" = as.numeric(dist_m_sclerotiourum[, 55]),
  "104" = as.numeric(dist_m_sclerotiourum[, 56]),
  "10" = as.numeric(dist_m_sclerotiourum[, 57]),
  "Al" = as.numeric(dist_m_sclerotiourum[, 58]),
  "Hi" = as.numeric(dist_m_sclerotiourum[, 59]),
  "In" = as.numeric(dist_m_sclerotiourum[, 60]),
  "mont" = as.numeric(dist_m_sclerotiourum[, 61]),
  "Sand" = as.numeric(dist_m_sclerotiourum[, 62]),
  "Sani" = as.numeric(dist_m_sclerotiourum[, 63]),
  "St" = as.numeric(dist_m_sclerotiourum[, 64]),
  "howard" = as.numeric(dist_m_sclerotiourum[, 65]),
  "chi" = as.numeric(dist_m_sclerotiourum[, 66]),
  "floyd" = as.numeric(dist_m_sclerotiourum[, 67]),
  "sto" = as.numeric(dist_m_sclerotiourum[, 68]),
  "97" = as.numeric(dist_m_sclerotiourum[, 69]),
  "be" = as.numeric(dist_m_sclerotiourum[, 70]),
  "ta" = as.numeric(dist_m_sclerotiourum[, 71]),
  "wau" = as.numeric(dist_m_sclerotiourum[, 72]),
  "la" = as.numeric(dist_m_sclerotiourum[, 73]),
  "co" = as.numeric(dist_m_sclerotiourum[, 74]),
  "cern" = as.numeric(dist_m_sclerotiourum[, 75]),
  "cer" = as.numeric(dist_m_sclerotiourum[, 76])
  )

# Replace 0 with NA
#sclerotiourum_dist_matrix[sclerotiourum_dist_matrix == 0] <- NA              
# object to hierarchially cluster 
hc_sclerotiourum <- hclust(as.dist(dist_m_sclerotiourum), method="complete")
# 50 km of distance for each one
d <- 50000

sclerotiourum_dist_matrix$cluster <- cutree(hc_sclerotiourum, h=d)
plot(dist_matrix$cluster)


```

###Subsetting some data
```{r, warning=FALSE}
# Creating the base map for subsequent plots
mid_west_base <-
  ggplot(data = mid_west_county,
         mapping = aes(x = long, y = lat, group = group)) +
  coord_fixed(1.3) +
  geom_polygon(color = "black", fill = "grey")
mexico_base <-
  ggplot(data = mexico,
         mapping = aes(x = long, y = lat, group = group)) +
  coord_fixed(1.3) +
  geom_polygon(color = "black", fill = "grey") + coord_map(projection = "mercator",xlim = c(-110, -107), ylim = c(24, 27))
# Creating the Master Dataframe about selection
selection <-
  wm1_ %>% select(# selecting the most important 15 columns plus the new one created
    Collection_ID,
    State,
    County,
    Field,
    Year,
    Origin,
    long,
    lat,
    hyphal_tip..ht.,
    DNA.Extraction,
    Serial_agar_dilution_for_creating_the_model_..N.20.,
    Form.ID,
    Fungicide_status,
    num_isolates_per_field
  ) %>% 
  arrange(factor (State, levels = c("NE", "IA", "WI", "MI"))) %>%#Arrange in this order from left to right in the US map  
  group_by(State, County, Field) %>%# grouping in order to select those with >=3 isolates/field/County/State 
  mutate(num_ISO_per_field = n()) %>%# Creating the column to count them 
  ungroup %>%
  group_by(State, County) %>%# grouping in order to select those   fields(>=3 isolates)/County/State 
  mutate(num_field_County = length(unique(factor(Field)))) %>%# Creating the column to count them 
  select(# Ordering the collumns to easy visulalization
    Collection_ID,
    State,
    County,
    Field,
    num_ISO_per_field,
    num_field_County,
    Year,
    Origin,
    long,
    lat,
    hyphal_tip..ht.,
    DNA.Extraction,
    Serial_agar_dilution_for_creating_the_model_..N.20.,
    Form.ID,
    Fungicide_status,
    num_isolates_per_field
  )
#Subdata for randomly selection of those 3 isolates/field/county/state 
selection1 <- selection %>%
  ungroup() %>%
  group_by(State, County, Field) %>%#grouping in order to select 3 isolates/field/County/State 
  filter(num_ISO_per_field > 10) %>%# Filtering those fields with more than 10 isolates/ field . Here the original idea was to filter by 3 >= isolates but  it does not not work with the next pipe available from tidyverse 
  sample_n(3, replace = F) #Randomly sampled 3 isolates from the 10 available from each group 

# Subdata another one to work with those groups with less than 10 isolates and more or equal to 3 isolates. The way to do it was to rest to the ones done 
selection2 <- selection %>%# using the master dataframe
  ungroup() %>%
  group_by(State, County, Field) %>%#grouping in order to select 3 isolates/field/County/State 
  filter(!num_ISO_per_field %in% selection1$num_ISO_per_field)%>%#filterring those already done 
  ungroup()

```
###Finding randomly either by function or by hand writing  3 samples for each (by sample function) field for those groupings with less than 10 isolates/field. In those cases of error it is due there is less than 3 isolates per field but it still being okay 

```{r}
fields1 <- selection2 %>%
  filter(State == "NE", County == "Valley", Field == "va")
a <- sample(fields1$Collection_ID, size = 3, replace = FALSE)# sampling 3 for each group

fields2 <- selection2 %>%
  filter(State == "NE", County == "Antelope", Field == "ant")
b <- sample(fields2$Collection_ID, size = 3, replace = FALSE)

fields3 <- selection2 %>%
  filter(State == "NE", County == "Antelope", Field == "21")
c <- sample(fields3$Collection_ID, size = 3, replace = FALSE)

fields4 <- selection2 %>%
  filter(State == "NE", County == "Antelope", Field == "22")
d <- sample(fields4$Collection_ID, size = 3, replace = FALSE)

fields5 <- selection2 %>%
  filter(State == "NE", County == "Antelope", Field == "23")
e <- sample(fields5$Collection_ID, size = 3, replace = FALSE)

fields6 <- selection2 %>%
  filter(State == "NE", County == "Antelope", Field == "24")
f <- sample(fields6$Collection_ID, size = 3, replace = FALSE)

fields7 <- selection2 %>%
  filter(State == "NE", County == "Antelope", Field == "25")
g <- sample(fields7$Collection_ID, size = 3, replace = FALSE)

fields8 <- selection2 %>%
  filter(State == "NE", County == "Antelope", Field == "69")
h <- sample(fields8$Collection_ID, size = 3, replace = FALSE)

fields9 <- selection2 %>%
  filter(State == "NE", County == "Antelope", Field == "75")
i <- sample(fields9$Collection_ID, size = 3, replace = FALSE)

fields10 <- selection2 %>%
  filter(State == "NE", County == "Antelope", Field == "102")
#j <- sample(fields10$Collection_ID, size = 3, replace = FALSE) less than 3 isolates per field

fields11 <- selection2 %>%
  filter(State == "NE", County == "Antelope", Field == "103")
#k <- sample(fields11$Collection_ID, size = 3, replace = FALSE) less than 3 isolates per field

fields12 <- selection2 %>%
  filter(State == "NE", County == "Cedar", Field == "6")
l <- sample(fields12$Collection_ID, size = 3, replace = FALSE)

fields13 <- selection2 %>%
  filter(State == "NE", County == "Cedar", Field == "67")
m <- sample(fields13$Collection_ID, size = 3, replace = FALSE)

fields14 <- selection2 %>%
  filter(State == "NE", County == "Cedar", Field == "78")
n <- sample(fields14$Collection_ID, size = 3, replace = FALSE)

fields15 <- selection2 %>%
  filter(State == "NE", County == "Cedar", Field == "98")
o <- sample(fields15$Collection_ID, size = 3, replace = FALSE)

fields16 <- selection2 %>%
  filter(State == "NE", County == "Cedar", Field == "100")
p <- sample(fields16$Collection_ID, size = 3, replace = FALSE)

fields17 <- selection2 %>%
  filter(State == "NE", County == "Cedar", Field == "101")
q <- sample(fields17$Collection_ID, size = 3, replace = FALSE)

fields18 <- selection2 %>%
  filter(State == "NE", County == "Greeley", Field == "8")
r <- sample(fields18$Collection_ID, size = 3, replace = FALSE)

fields19 <- selection2 %>%
  filter(State == "NE", County == "Greeley", Field == "9")
s <- sample(fields19$Collection_ID, size = 3, replace = FALSE)

fields20 <- selection2 %>%
  filter(State == "NE", County == "Greeley", Field == "68")
#t <- sample(fields20$Collection_ID, size = 3, replace = FALSE) less than 3 isolates per field

fields21 <- selection2 %>%
  filter(State == "NE", County == "Madison", Field == "10")
u <- sample(fields21$Collection_ID, size = 3, replace = FALSE)

fields22 <- selection2 %>%
  filter(State == "NE", County == "Madison", Field == "11")
v <- sample(fields22$Collection_ID, size = 3, replace = FALSE)

fields23 <- selection2 %>%
  filter(State == "NE", County == "Boone", Field == "18")
w <- sample(fields23$Collection_ID, size = 3, replace = FALSE)

fields24 <- selection2 %>%
  filter(State == "NE", County == "Boone", Field == "19")
x <- sample(fields24$Collection_ID, size = 3, replace = FALSE)

fields25 <- selection2 %>%
  filter(State == "NE", County == "Boone", Field == "20")
y <- sample(fields25$Collection_ID, size = 3, replace = FALSE)

fields26 <- selection2 %>%
  filter(State == "NE", County == "Boone", Field == "45")
z <- sample(fields26$Collection_ID, size = 3, replace = FALSE)

fields27 <- selection2 %>%
  filter(State == "NE", County == "Boone", Field == "46")
aa <- sample(fields27$Collection_ID, size = 3, replace = FALSE)

fields28 <- selection2 %>%
  filter(State == "NE", County == "Boone", Field == "48")
bb <- sample(fields28$Collection_ID, size = 3, replace = FALSE)

fields29 <- selection2 %>%
  filter(State == "NE", County == "Dodge", Field == "34")
cc <- sample(fields29$Collection_ID, size = 3, replace = FALSE)

fields30 <- selection2 %>%
  filter(State == "NE", County == "Dodge", Field == "35")
dd <- sample(fields30$Collection_ID, size = 3, replace = FALSE)

fields31 <- selection2 %>%
  filter(State == "NE", County == "Holt", Field == "47")
ee <- sample(fields31$Collection_ID, size = 3, replace = FALSE)

fields32 <- selection2 %>%
  filter(State == "NE", County == "Sherman", Field == "60")
ff <- sample(fields32$Collection_ID, size = 3, replace = FALSE)

fields33 <- selection2 %>%
  filter(State == "NE", County == "Kearney", Field == "62")
gg <- sample(fields33$Collection_ID, size = 3, replace = FALSE)

fields34 <- selection2 %>%
  filter(State == "NE", County == "Kearney", Field == "104")
#hh <- sample(fields34$Collection_ID, size = 3, replace = FALSE) less than 3 isolates per field

fields35 <- selection2 %>%
  filter(State == "NE", County == "Kearney", Field == "10")
ii <- sample(fields35$Collection_ID, size = 3, replace = FALSE)

fields36 <- selection2 %>%
  filter(State == "NE", County == "Cuming", Field == "66")
#jj <- sample(fields36$Collection_ID, size = 3, replace = FALSE) less than 3 isolates per field

fields37 <- selection2 %>%
  filter(State == "NE", County == "Pierce", Field == "97")
kk <-  sample(fields37$Collection_ID, size = 3, replace = FALSE)# less than 3 isolates per field

fields38 <- selection2 %>%
  filter(State == "NE", County == "Knox", Field == "99")
ll <- sample(fields38$Collection_ID, size = 3, replace = FALSE)

fields39 <- selection2 %>%
  filter(State == "IA", County == "Howard", Field == "howard")
# mm <- sample(fields39$Collection_ID, size = 3, replace = FALSE) less than 3 isolates per field

fields40 <- selection2 %>%
  filter(State == "IA", County == "Chickasaw", Field == "chi")
# nn <- sample(fields40$Collection_ID, size = 3, replace = FALSE) less than 3 isolates per field

fields41 <- selection2 %>%
  filter(State == "IA", County == "Floyd", Field == "floyd")
# oo <- sample(fields41$Collection_ID, size = 3, replace = FALSE) less than 3 isolates per field

fields42 <- selection2 %>%
  filter(State == "IA", County == "Story", Field == "sto")
pp <- sample(fields42$Collection_ID, size = 3, replace = FALSE)

fields43 <- selection2 %>%
  filter(State == "IA", County == "Shelby", Field == "sh")
qq <- sample(fields43$Collection_ID, size = 3, replace = FALSE)

fields44 <- selection2 %>%
  filter(State == "IA", County == "Benton", Field == "be")
rr <- sample(fields44$Collection_ID, size = 3, replace = FALSE)

fields45 <- selection2 %>%
  filter(State == "IA", County == "Tama", Field == "ta")
ss <- sample(fields45$Collection_ID, size = 3, replace = FALSE)

fields46 <- selection2 %>%
  filter(State == "WI", County == "Sauk City", Field == "sa")
tt <- sample(fields46$Collection_ID, size = 3, replace = FALSE)

fields47 <- selection2 %>%
  filter(State == "WI", County == "Cuba City", Field == "cu")
uu <- sample(fields47$Collection_ID, size = 3, replace = FALSE)

fields48 <- selection2 %>%
  filter(State == "WI", County == "Waushara", Field == "wau")
vv <- sample(fields48$Collection_ID, size = 3, replace = FALSE)

fields49 <- selection2 %>%
  filter(State == "WI", County == "Grant", Field == "la")
ww <- sample(fields49$Collection_ID, size = 3, replace = FALSE)

fields50 <- selection2 %>%
  filter(State == "WI", County == "Iowa", Field == "co")
# xx <- sample(fields50$Collection_ID, size = 3, replace = FALSE) less than 3 isolates per field

fields51 <- selection2 %>%
  filter(State == "WI", County == "Rock", Field == "cern")
# yy <- sample(fields51$Collection_ID, size = 3, replace = FALSE) less than 3 isolates per field

fields52 <- selection2 %>%
  filter(State == "WI", County == "Rock", Field == "cer")
zz <- sample(fields52$Collection_ID, size = 3, replace = FALSE)

fields53 <- selection2 %>%
  filter(State == "MI", County == "Allegan", Field == "Al")
aaa <- sample(fields53$Collection_ID, size = 3, replace = FALSE)

```
###Once all filters, add a new column called "Status" to the master mix dataframe 
```{r}
selection <- selection %>%
  mutate(
    Status = case_when(
      Collection_ID %in% selection1$Collection_ID ~ "selected",# filtering and naming "selected"" in case of in
      Collection_ID %in% a ~ "selected",
      Collection_ID %in% b ~ "selected",
      Collection_ID %in% c ~ "selected",
      Collection_ID %in% d ~ "selected",
      Collection_ID %in% e ~ "selected",
      Collection_ID %in% f ~ "selected",
      Collection_ID %in% g ~ "selected",
      Collection_ID %in% h ~ "selected",
      Collection_ID %in% i ~ "selected",
      Collection_ID %in% l ~ "selected",
      Collection_ID %in% m ~ "selected",
      Collection_ID %in% n ~ "selected",
      Collection_ID %in% o ~ "selected",
      Collection_ID %in% p ~ "selected",
      Collection_ID %in% q ~ "selected",
      Collection_ID %in% r ~ "selected",
      Collection_ID %in% s ~ "selected",
      Collection_ID %in% u ~ "selected",
      Collection_ID %in% v ~ "selected",
      Collection_ID %in% ww ~ "selected",
      Collection_ID %in% x ~ "selected",
      Collection_ID %in% y ~ "selected",
      Collection_ID %in% z ~ "selected",
      Collection_ID %in% aa ~ "selected",
      Collection_ID %in% bb ~ "selected",
      Collection_ID %in% cc ~ "selected",
      Collection_ID %in% dd ~ "selected",
      Collection_ID %in% ee ~ "selected",
      Collection_ID %in% ff ~ "selected",
      Collection_ID %in% gg ~ "selected",
      Collection_ID %in% ii ~ "selected",
      Collection_ID %in% kk ~ "selected",
      Collection_ID %in% ll ~ "selected",
      Collection_ID %in% pp ~ "selected",
      Collection_ID %in% rr ~ "selected",
      Collection_ID %in% ss ~ "selected",
      Collection_ID %in% tt ~ "selected",
      Collection_ID %in% uu ~ "selected",
      Collection_ID %in% vv ~ "selected",
      Collection_ID %in% ww ~ "selected",
      Collection_ID %in% zz ~ "selected",
      Collection_ID %in% aaa ~ "selected",
      TRUE ~ "not_selected"# in case of not, naming "not_selected""
    )
  )  

```

```{r}
  ##Summarizing selected by Widwest
selection_FINAL_objective1 <-
  selection %>% filter(Status == "selected" |
                         Collection_ID == "2344" | Collection_ID == "2403") %>%  ungroup() #selecting those that are far from the group less than 3 isolates/field 
 
selection_FINAL_objective1$Status[selection_FINAL_objective1$Collection_ID == "2344"] <- "selected" #renaming these for now selected
selection_FINAL_objective1$Status[selection_FINAL_objective1$Collection_ID == "2403"] <- "selected" #renaming these for now selected

selection_FINAL_objective1_filterring <-
 selection_FINAL_objective1 %>% filter(Fungicide_status == "no_tested") #by no fungicide pre-tested


```

#PLOTS  
##General Visualization of the US midewest isolates by different levels
```{r, echo=FALSE, warning= FALSE}
base <-
  mid_west_base + geom_point(# using basis of the previous subdata
    data = wm1_,
    aes(x = long, y = lat, color = Origin),
    inherit.aes = FALSE,
    size = 1
    # na.rm = TRUE,
    #  shape = 22,
    #  stroke = 0.5,
    #  fill = "white"
  ) + theme(legend.position = "bottom") + labs(title = "Sclerotinia sclerotiorum Soybean inventory from Midwest") + theme(plot.title = element_text(
    size = 10,
    face = "bold",
    hjust = 0.5,
    family = "Helvetica"
  ))
base

```

#Selection of the State of Nebraska  
###As we can noticed prevously for this state, there are no fungicide field trials, in other words, there are just farmer fields
```{r, warning= FALSE}
WM1S <- selection %>%# creating a subdata of NE
  group_by(State, Origin) %>%
  filter(Origin == "Farmer fields", State == "NE") %>%
  ungroup()

WM1S$DNA.Extraction = factor(WM1S$DNA.Extraction ,
                             levels = c("No DNA extraction", "Yes DNA extraction"))#Ordering the levels of DNA.Extraction
NE_county <- subset(counties, region == "nebraska")# Subsetting just for NE state

NE_county_base <-# Creating the base to use for next plots
  ggplot(data = NE_county,
         mapping = aes(x = long, y = lat, group = group)) +
  coord_fixed(1.3) +
  geom_polygon(color = "black", fill = "grey")

###filterring by the first Id in a field
WM1S_ <-
  WM1S %>% filter(
    Collection_ID == "1255"|
    Collection_ID == "1285"|
    Collection_ID == "1297"|
    Collection_ID == "1317"|
    Collection_ID == "1831"|
    Collection_ID == "1841"|
    Collection_ID == "1941"|
    Collection_ID == "1951"|    
    Collection_ID == "1961"|
    Collection_ID == "1966"|
    Collection_ID == "1971"|
    Collection_ID == "1981"|
    Collection_ID == "1991"|
    Collection_ID == "2001"|
    Collection_ID == "2006"|
    Collection_ID == "2013"|
    Collection_ID == "2017"|
    Collection_ID == "2022"|
    Collection_ID == "2027"|
    Collection_ID == "2037"|
    Collection_ID == "2047"|
    Collection_ID == "2057"|
    Collection_ID == "2067"|
    Collection_ID == "2077"|
    Collection_ID == "2087"|
    Collection_ID == "2097"|
    Collection_ID == "2107"|
    Collection_ID == "2108"|
    Collection_ID == "2118"|
    Collection_ID == "2119"|
    Collection_ID == "2129"|
    Collection_ID == "2139"|
    Collection_ID == "2149"|
    Collection_ID == "2159"|
    Collection_ID == "2169"|
    Collection_ID == "2179"|
    Collection_ID == "2189"|
    Collection_ID == "2199"|
    Collection_ID == "2200"|
    Collection_ID == "2201"|
    Collection_ID == "2203"
  )
#Plot of NE selection
blumeria <- NE_county_base + geom_point(
  data = WM1S,
  aes(
    x = long,
    y = lat,
    color = Year#by color the level of selection
    #shape = Year,# by shape the level of Year
    #size = DNA.Extraction# by size the level of DNA.Extraction
  ),
  inherit.aes = FALSE,
  na.rm = TRUE
) + theme(legend.position = "bottom") + #+ facet_wrap( ~ num_isolates_per_field)  #Wrapping by the # of isolates/field
  scale_color_manual(values = c("red", "blue")) + theme(legend.position = "bottom") +  labs(title = "Sclerotinia sclerotiorum Soybean  from Nebraska") + theme(plot.title = element_text(
    size = 10,
    face = "bold",
    hjust = 0.5,
    family = "Helvetica"
  ))
blumeria

#Plot of NE selection
face <- NE_county_base + geom_point(
  data = WM1S,
  aes(
    x = long,
    y = lat,
    color = num_isolates_per_field #by color the level of selection
    #shape = Year,# by shape the level of Year
    #size = DNA.Extraction# by size the level of DNA.Extraction
  ),
  inherit.aes = FALSE,
  na.rm = TRUE
) + theme(legend.position = "bottom") + #+ facet_wrap( ~ num_isolates_per_field)  #Wrapping by the # of isolates/field
  scale_color_manual(values = c("red", "blue")) + theme(legend.position = "bottom") +  labs(title = "Sclerotinia sclerotiorum Soybean  from Nebraska") + theme(plot.title = element_text(
    size = 10,
    face = "bold",
    hjust = 0.5,
    family = "Helvetica"
  ))
face


```


```{r, echo=FALSE, warning= FALSE}
WM1S <- selection %>%# creating a subdata of NE
  group_by(State, Origin) %>%
  filter(Origin == "Farmer fields", State == "NE") %>%
  ungroup()

WM1S$DNA.Extraction = factor(WM1S$DNA.Extraction ,
                             levels = c("No DNA extraction", "Yes DNA extraction"))#Ordering the levels of DNA.Extraction
NE_county <- subset(counties, region == "nebraska")# Subsetting just for NE state

NE_county_base <-# Creating the base to use for next plots
  ggplot(data = NE_county,
         mapping = aes(x = long, y = lat, group = group)) +
  coord_fixed(1.3) +
  geom_polygon(color = "black", fill = "grey")

#Plot of NE selection
# face <- NE_county_base + geom_point(
#   data = WM1S,
#   aes(
#     x = long,
#     y = lat,
#     color = Selected, #by color the level of selection
#     shape = Year,# by shape the level of Year
#     size = DNA.Extraction# by size the level of DNA.Extraction
#   ),
#   inherit.aes = FALSE,
#   na.rm = TRUE
# ) + theme(legend.position = "bottom") + facet_wrap( ~ num_isolates_per_field) + #Wrapping by the # of isolates/field
#   scale_color_manual(values = c("red", "blue")) + theme(legend.position = "bottom") + labs(title = "Sclerotinia sclerotiorum Soybean selection from Nebraska") + theme(plot.title = element_text(
#     size = 10,
#     face = "bold",
#     hjust = 0.5,
#     family = "Helvetica"
#   ))
# face

# As we can noticed for DNA extraction here, it's been used those furthest places

# Now closer sight by counties
NE_counties <-
  counties %>% group_by(region) %>% filter(region == "nebraska") %>% ungroup %>% filter(#filterring by those counties with more fields
    subregion == "antelope" |
      subregion == "madison" |
      subregion == "cedar" |
      subregion == "cuming" |
      subregion == "dodge" |
      subregion == "boone" |
      subregion == "holt" |
      subregion == "valley" |
      subregion == "greeley" |
      subregion == "sherman" |
      subregion == "howard"
  )

# new object created in order to get the center of the polygon and place there the name of each county
centroid <-
  aggregate(data = NE_counties, cbind(long, lat) ~ subregion, FUN = mean)# getting the average

#Plotting the polygon using the as basis the centroid object created previously
NE_counties_base <-  ggplot(data = NE_counties,
                            mapping = aes(x = long, y = lat, group = subregion)) +# here the group means the counties
  coord_fixed(1.3) +
  geom_polygon(color = "black", fill = "grey") +  geom_text(
    data = centroid,
    aes(label = subregion, x = long, y = lat),
    check_overlap = TRUE,
    size = 3
  )
#Plot of NE selection with counties delimitation
hand <- NE_counties_base + geom_point(
  data = WM1S,
  aes(
    x = long,
    y = lat,
    color = Status 
    #label= Collection_ID
    # by color the level of selection
    #shape = Year# by shape the level of Year
    #size = DNA.Extraction# by size the level of DNA.Extraction
  ),
  inherit.aes = FALSE,
  na.rm = TRUE
) + theme(legend.position = "bottom") + facet_wrap( ~ num_isolates_per_field) + #Wrapping by the # of isolates/field
  scale_color_manual(values = c("red", "blue")) + theme(legend.position = "bottom") + labs(title = "Sclerotinia sclerotiorum Soybean selection from Nebraska") + theme(plot.title = element_text(
    size = 10,
    face = "bold",
    hjust = 0.5,
    family = "Helvetica"
  )) 

hand

```

#Selection of the State of Iowa  
```{r echo=FALSE, warning= FALSE,}
WM_IA <- selection %>%# creating a subdata of IA
  filter(State == "IA")

WM_IA$DNA.Extraction = factor(WM_IA$DNA.Extraction ,
                             levels = c("No DNA extraction", "Yes DNA extraction"))#Ordering the levels of DNA.Extraction


WM_IA$Status = factor(WM_IA$Fungicide_status , levels = c("tested", "no_tested"))#Ordering the levels of Fungicide pre-tested

NE_county_base <-# Subsetting just for NE state
IA_county <- subset(counties, region == "iowa")# Creating the base to use for next plots
IA_county_base <-
  ggplot(data = IA_county,
         mapping = aes(x = long, y = lat, group = group)) +
  coord_fixed(1.3) +
  geom_polygon(color = "black", fill = "white")

#Plot of IA selection
eye <- IA_county_base  + geom_point(
  data = WM_IA,
  aes(
    x = long,
    y = lat,
    color = num_isolates_per_field# by color the level of selection
    #shape = Year,# by shape the level of Year
    #size = DNA.Extraction# by size the level of DNA.Extraction
  ),
  inherit.aes = FALSE,
  na.rm = TRUE
) + theme(legend.position = "bottom") + #facet_wrap( ~ num_isolates_per_field) +#Wrapping by the # of isolates/field
  scale_color_manual(values = c("red", "blue")) + theme(legend.position = "bottom") + labs(title = "Sclerotinia sclerotiorum Soybean  from Iowa") + theme(plot.title = element_text(
    size = 10,
    face = "bold",
    hjust = 0.5,
    family = "Helvetica"
  ))
eye

```

```{r, echo=FALSE, warning= FALSE}
WM_IA <- selection %>%# creating a subdata of IA
  filter(State == "IA")

WM_IA$DNA.Extraction = factor(WM_IA$DNA.Extraction ,
                             levels = c("No DNA extraction", "Yes DNA extraction"))#Ordering the levels of DNA.Extraction


WM_IA$Status = factor(WM_IA$Fungicide_status , levels = c("tested", "no_tested"))#Ordering the levels of Fungicide pre-tested

NE_county_base <-# Subsetting just for NE state
IA_county <- subset(counties, region == "iowa")# Creating the base to use for next plots
IA_county_base <-
  ggplot(data = IA_county,
         mapping = aes(x = long, y = lat, group = group)) +
  coord_fixed(1.3) +
  geom_polygon(color = "black", fill = "white")

#Plot of IA selection
# eye <- IA_county_base  + geom_point(
#   data = WM_IA,
#   aes(
#     x = long,
#     y = lat,
#     color = Status,# by color the level of selection
#     shape = Year,# by shape the level of Year
#     size = DNA.Extraction# by size the level of DNA.Extraction
#   ),
#   inherit.aes = FALSE,
#   na.rm = TRUE
# ) + theme(legend.position = "bottom") + facet_wrap( ~ num_isolates_per_field) +#Wrapping by the # of isolates/field
#   scale_color_manual(values = c("red", "blue")) + theme(legend.position = "bottom") + labs(title = "Sclerotinia sclerotiorum Soybean inventory from Iowa") + theme(plot.title = element_text(
#     size = 10,
#     face = "bold",
#     hjust = 0.5,
#     family = "Helvetica"
#   ))
# eye
# 

# As we can noticed for DNA extraction here, it's been used those furthest places
# Now closer sight by counties

IA_counties <-
  counties %>% group_by(region) %>% filter(region == "iowa") %>% ungroup %>% filter(#filtering by those counties with more fields
    subregion == "floyd" |
      subregion == "butler" |
      subregion == "chickasaw" |
      subregion == "shelby" |
      subregion == "boone" |
      subregion == "polk" |
      subregion == "bremer" | subregion == "benton" | subregion == "tama"
  )
# new object created in order to get the center of the polygon and place there the name of each county
centroid_iowa <-
  aggregate(data = IA_counties, cbind(long, lat) ~ subregion, FUN = mean)

#Plotting the polygon using the as basis the centroid object created previously
IA_counties_base <-  ggplot(data = IA_counties,
                            mapping = aes(x = long, y = lat, group = subregion)) +# here the group means the counties
  coord_fixed(1.3) +
  geom_polygon(color = "black", fill = "grey") +  geom_text(
    data = centroid_iowa,
    aes(label = subregion, x = long, y = lat),
    check_overlap = TRUE,
    size = 3
  )

#Plot of IA selection with counties delimitation
pinky <- IA_counties_base + geom_point(
  data = WM_IA,
  aes(
    x = long,
    y = lat,
    color = Status# by color the level of selection
    #shape = Year,# by shape the level of Year
   # size = DNA.Extraction# by size the level of DNA.Extraction
  ),
  inherit.aes = FALSE,
  na.rm = TRUE
) + theme(legend.position = "bottom") + facet_wrap( ~ num_isolates_per_field) +#Wrapping by the # of isolates/field
  scale_color_manual(values = c("red", "blue")) + theme(legend.position = "bottom") + labs(title = "Sclerotinia sclerotiorum Soybean inventory from Nebraska 'No fungicide tested'") + theme(plot.title = element_text(
    size = 10,
    face = "bold",
    hjust = 0.5,
    family = "Helvetica"
  ))
pinky

```

#Selection of the State of Wisconsin

```{r, echo=FALSE, warning= FALSE}
WM_WI <- selection %>%# creating a subdata of WI
  filter(State == "WI")
#Ordering the levels of DNA.Extraction
WM_IA$DNA.Extraction = factor(WM_IA$DNA.Extraction ,
                             levels = c("No DNA extraction", "Yes DNA extraction"))

WM_WI$Status = factor(WM_WI$Fungicide_status , levels = c("tested", "no_tested"))#Ordering the levels of Fungicide pre-tested

# Subsetting just for WI state
WM_WIcounty <- subset(counties, region == "wisconsin")
WI_county_base <-# Creating the base to use for next plots
  ggplot(data = WM_WIcounty,
         mapping = aes(x = long, y = lat, group = group)) +
  coord_fixed(1.3) +
  geom_polygon(color = "black", fill = "white")
#Plot of WI selection
nose <- WI_county_base  + geom_point(
  data = WM_WI,
  aes(
    x = long,
    y = lat,
    color = num_isolates_per_field# by color the level of selection
    #shape = Year,# by shape the level of Year
    #size = DNA.Extraction# by size the level of DNA.Extraction
  ),
  inherit.aes = FALSE,
  na.rm = TRUE
) + theme(legend.position = "bottom") + #facet_wrap( ~ num_isolates_per_field) +#Wrapping by the # of isolates/field
  scale_color_manual(values = c("red", "blue")) + theme(legend.position = "bottom") + labs(title = "Sclerotinia sclerotiorum Soybean from Wisconsin") + theme(plot.title = element_text(
    size = 10,
    face = "bold",
    hjust = 0.5,
    family = "Helvetica"
  ))
nose


```

```{r, echo=FALSE, warning= FALSE}
WM_WI <- selection %>%# creating a subdata of WI
  filter(State == "WI")
#Ordering the levels of DNA.Extraction
WM_IA$DNA.Extraction = factor(WM_IA$DNA.Extraction ,
                             levels = c("No DNA extraction", "Yes DNA extraction"))

WM_WI$Status = factor(WM_WI$Fungicide_status , levels = c("tested", "no_tested"))#Ordering the levels of Fungicide pre-tested

# Subsetting just for WI state
WM_WIcounty <- subset(counties, region == "wisconsin")
WI_county_base <-# Creating the base to use for next plots
  ggplot(data = WM_WIcounty,
         mapping = aes(x = long, y = lat, group = group)) +
  coord_fixed(1.3) +
  geom_polygon(color = "black", fill = "white")
#Plot of WI selection
# nose <- WI_county_base  + geom_point(
#   data = WM_WI,
#   aes(
#     x = long,
#     y = lat,
#     color = Status,# by color the level of selection
#     shape = Year,# by shape the level of Year
#     size = DNA.Extraction# by size the level of DNA.Extraction
#   ),
#   inherit.aes = FALSE,
#   na.rm = TRUE
# ) + theme(legend.position = "bottom") + facet_wrap( ~ num_isolates_per_field) +#Wrapping by the # of isolates/field
#   scale_color_manual(values = c("red", "blue")) + theme(legend.position = "bottom") + labs(title = "Sclerotinia sclerotiorum Soybean selection from Wisconsin") + theme(plot.title = element_text(
#     size = 10,
#     face = "bold",
#     hjust = 0.5,
#     family = "Helvetica"
#   ))
# nose

# As we can noticed for DNA extraction here, it's been used those furthest places
# Now closer sight by countiesWI_counties <-

WI_counties <-
  counties %>% group_by(region) %>% filter(region == "wisconsin") %>% ungroup %>% filter(#filtering by those counties with more fields
    subregion == "walworth" |
      subregion == "grant" |
      subregion == "lafayette" |
      subregion == "waushara" | subregion == "adams" | subregion == "dane"
  )
 
  # new object created in order to get the center of the polygon and place there the name of each county
centroid_wisconsin <-
  aggregate(data = WI_counties, cbind(long, lat) ~ subregion, FUN = mean)

#Plotting the polygon using the as basis the centroid object created previously
WI_counties_base <-  ggplot(data = WI_counties,
                            mapping = aes(x = long, y = lat, group = subregion)) +# here the group means the counties
  coord_fixed(1.3) +
  geom_polygon(color = "black", fill = "grey") +  geom_text(
    data = centroid_wisconsin,
    aes(label = subregion, x = long, y = lat),
    check_overlap = TRUE,
    size = 3
  )
#Plot of WI selection with counties delimitation
pointfinger <- WI_counties_base + geom_point(
  data = WM_WI,
  aes(
    x = long,
    y = lat,
    color = Status# by color the level of selection
    #shape = Year,# by shape the level of Year
    #size = DNA.Extraction# by size the level of DNA.Extraction
  ),
  inherit.aes = FALSE,
  na.rm = TRUE
) + theme(legend.position = "bottom") + facet_wrap( ~ num_isolates_per_field) +#Wrapping by the # of isolates/field
  scale_color_manual(values = c("red", "blue")) + theme(legend.position = "bottom") + labs(title = "Sclerotinia sclerotiorum Soybean inventory from Wisconsin 'No fungicide tested'") + theme(plot.title = element_text(
    size = 10,
    face = "bold",
    hjust = 0.5,
    family = "Helvetica"
  ))
pointfinger

```

#Selection of the State of Michigan
```{r}
#Michigan#
WM_MI <- selection %>%# creating a subdata of MI
  filter(State == "MI")
#Ordering the levels of DNA.Extraction
WM_MI$DNA.Extraction = factor(WM_MI$DNA.Extraction ,
                             levels = c("No DNA extraction", "Yes DNA extraction"))#Ordering the levels of DNA.Extraction

WM_MI$Status = factor(WM_MI$Fungicide_status , levels = c("tested", "no_tested"))#Ordering the levels of Fungicide pre-tested

# Subsetting just for MI state
WM_MIcounty <- subset(counties, region == "michigan")
WM_county_base <-# Creating the base to use for next plots
  ggplot(data = WM_MIcounty,
         mapping = aes(x = long, y = lat, group = group)) +
  coord_fixed(1.3) +
  geom_polygon(color = "black", fill = "white")

#Plot of NE selection
mouth <- WM_county_base  + geom_point(
  data = WM_MI,
  aes(
    x = long,
    y = lat,
    color = num_isolates_per_field# by color the level of selection
    #shape = Year,# by shape the level of Year
    #size = DNA.Extraction# by size the level of DNA.Extraction
  ),
  inherit.aes = FALSE,
  na.rm = TRUE
) + theme(legend.position = "bottom") + #facet_wrap( ~ num_isolates_per_field) +#Wrapping by the # of isolates/field
  scale_color_manual(values = c("red", "blue")) + theme(legend.position = "bottom") + labs(title = "Sclerotinia sclerotiorum Soybean inventory from Iowa") + theme(plot.title = element_text(
    size = 10,
    face = "bold",
    hjust = 0.5,
    family = "Helvetica"
  ))
mouth


```

```{r, echo=FALSE, warning= FALSE}
#Michigan#
WM_MI <- selection %>%# creating a subdata of MI
  filter(State == "MI")
#Ordering the levels of DNA.Extraction
WM_MI$DNA.Extraction = factor(WM_MI$DNA.Extraction ,
                             levels = c("No DNA extraction", "Yes DNA extraction"))#Ordering the levels of DNA.Extraction

WM_MI$Status = factor(WM_MI$Fungicide_status , levels = c("tested", "no_tested"))#Ordering the levels of Fungicide pre-tested

# Subsetting just for MI state
WM_MIcounty <- subset(counties, region == "michigan")
WM_county_base <-# Creating the base to use for next plots
  ggplot(data = WM_MIcounty,
         mapping = aes(x = long, y = lat, group = group)) +
  coord_fixed(1.3) +
  geom_polygon(color = "black", fill = "white")

#Plot of NE selection
# mouth <- WM_county_base  + geom_point(
#   data = WM_MI,
#   aes(
#     x = long,
#     y = lat,
#     color = Status,# by color the level of selection
#     shape = Year,# by shape the level of Year
#     size = DNA.Extraction# by size the level of DNA.Extraction
#   ),
#   inherit.aes = FALSE,
#   na.rm = TRUE
# ) + theme(legend.position = "bottom") + facet_wrap( ~ num_isolates_per_field) +#Wrapping by the # of isolates/field
#   scale_color_manual(values = c("red", "blue")) + theme(legend.position = "bottom") + labs(title = "Sclerotinia sclerotiorum Soybean inventory from Iowa") + theme(plot.title = element_text(
#     size = 10,
#     face = "bold",
#     hjust = 0.5,
#     family = "Helvetica"
#   ))
# mouth

# As we can noticed for DNA extraction here, it's been used those furthest places
# Now closer sight by counties

MI_counties <-
  counties %>% group_by(region) %>% filter(region == "michigan") %>% ungroup %>% filter(#filtering by those counties with more fields
    subregion == "montcalm" |
      subregion == "ingham" |
      subregion == "allegan" |
      subregion == "ionia" |
      subregion == "joseph" | subregion == "hillsdale" |
      subregion == "sanilac"
  )
# new object created in order to get the center of the polygon and place there the name of each county
centroid_michigan <-
  aggregate(data = MI_counties, cbind(long, lat) ~ subregion, FUN = mean)

#Plotting the polygon using the as basis the centroid object created previously
MI_counties_base <-  ggplot(data = MI_counties,
                            mapping = aes(x = long, y = lat, group = subregion)) +# here the group means the counties
  coord_fixed(1.3) +
  geom_polygon(color = "black", fill = "grey") +  geom_text(
    data = centroid_michigan,
    aes(label = subregion, x = long, y = lat),
    check_overlap = TRUE,
    size = 3
  )
#Plot of MI selection with counties delimitation
anularfinger <- MI_counties_base  + geom_point(
  data = WM_MI,
  aes(
    x = long,
    y = lat,
    color = Status# by color the level of selection
    #shape = Year,# by shape the level of Year
    #size = DNA.Extraction# by size the level of DNA.Extraction
  ),
  inherit.aes = FALSE,
  na.rm = TRUE
) + theme(legend.position = "bottom") + facet_wrap( ~ num_isolates_per_field) + #Wrapping by the # of isolates/field
  scale_color_manual(values = c("red", "blue")) + theme(legend.position = "bottom") + labs(title = "Sclerotinia sclerotiorum Soybean inventory from Nebraska 'No fungicide tested'") + theme(plot.title = element_text(
    size = 10,
    face = "bold",
    hjust = 0.5,
    family = "Helvetica"
  ))
anularfinger

```




#SECOND PART OF DISSERTATION, Evaluate the effect of successive fungicide applications on sensitivity of populations of _S. sclerotiorum_ in field plots

#Bckground
IPM strategies for plant diseases are designed to reduce pesticide rates (Beckerman 2015). Low pesticide rates reduce selection pressure of major single gene resistance and multifactorial resistance (Gressel 1995). However, it is known that application of pesticides is necessarily non-uniform, which can result in sublethal exposure of the target pathogen to fungicides that may cause cellular stress. Such stress simulated under in vitro conditions was recently shown to increase mutation rates at SSR loci and alter genomic AFLP patterns (Amaradasa and Everhart 2016), and other studies have shown cellular stress can induce gene duplications (Kondrashov and Kondrashov 2006). Based on a model of evolutionary potential of pathogen populations, _S. sclerotiorum_ is considered to have intermediate risk of fungicide resistance (McDonald and Linde 2002). Population genetic studies of _S. sclerotiorum_ show low genotypic diversity and mostly clonal reproduction, which means that mutation is a more important source of genetic variation and may be related to fungicide resistance emergence. To test this hypothesis, the effect of field-applied fungicide doses on stress-induced mutation rate and fungicide sensitivity will be evaluated using simple sequence repeats (SSR) genotyping and fungicide sensitivity assays. Controls with no fungicide application will be compared to fungicide-treated plots.  
__the hypothesis of this objective is that fungicide field applications ( stress-induced mutation) increase genetic diversity of _S. sclerotiorum_ population.__ 

##General Visualization of the US midewest isolates from fungicide field trials at different years


```{r}
#Subdata by "Farmer fields""
# WMfarmerfields <- wm1_ %>%
#   group_by(State, Year, Origin) %>%
#   filter(Origin == "Farmer fields")
# 
# base2 <-
#   mid_west_base + geom_point(# using basis of the previous subdata
#     data = WMfarmerfields,
#     aes(x = long, y = lat, color = Year),
#     inherit.aes = FALSE,
#     size = 1,
#     na.rm = TRUE,
#     shape = 22,
#     stroke = 0.5,
#     fill = "white"
#   ) + theme(legend.position = "bottom") + labs(title = "Sclerotinia sclerotiorum of Farmer Field Soybean from Midwest") + theme(plot.title = element_text(  size = 10,
#   face = "bold",
#   hjust = 0.5,
#   family = "Helvetica"
# ))
# base2

#Subdata by Fungicide field trials
WMfungicide <- wm1_ %>%
group_by(State, Year, Origin) %>%
filter(Origin == "Fungicide field trials") %>% ungroup  

  base3 <-
mid_west_base + geom_point(# using basis of the previous subdata
data = WMfungicide,
aes(x = long, y = lat, color = Year),
inherit.aes = FALSE,
size = 1,
na.rm = TRUE
) + theme(legend.position = "bottom") + labs(title = "Sclerotinia sclerotiorum of Fungicide Field Trials Soybean from Midwest") + theme(plot.title = element_text(
size = 10,
face = "bold",
hjust = 0.5,
family = "Helvetica"
))
base3
WMfungicide <- wm1_ %>%
group_by(State, Year, Origin) %>%
filter(Origin == "Fungicide field trials")
base3 <-
mid_west_base + geom_point(# using basis of the previous subdata
data = WMfungicide,
aes(x = long, y = lat, color = Year),
inherit.aes = FALSE,
size = 2,
na.rm = TRUE
) + theme(legend.position = "bottom") + labs(title = "Sclerotinia sclerotiorum of Fungicide Field Trials Soybean from Midwest") + theme(plot.title = element_text(
size = 10,
face = "bold",
hjust = 0.5,
family = "Helvetica"
))
base3

```


##Objective 2 
  Select isolates by the time,  by filterring the points from IA & MI before visualized, make a Venn diagram with all already used for fungicide trials from objective 1.
 
```{r}
IA_fungicide <- WMfungicide %>% filter(State=="IA",  County =="Floyd"|County =="Chickasaw"|County =="Butler"|County =="Bremer")
MI_fungicide <- WMfungicide %>%ungroup() %>%  filter(  Field =="mon" |Field =="mont"  )
montcalm <- c("1671":"1830",  "2247":"2302")
#fungicideTRIALS <- MI_fungicide %>% bind_rows(MI_fungicide)
bettle <- MI_fungicide %>% 
mutate(Fungicide_status = ifelse(Collection_ID%in%selection_FINAL_objective1$Collection_ID,
                                 "sync",ifelse(Collection_ID%in% tested,"sync","no_sync"))) %>% filter(Fungicide_status== "sync")
aphelenchus <-IA_fungicide %>% 
mutate(Fungicide_status = ifelse(Collection_ID%in%selection_FINAL_objective1$Collection_ID,
                                 "sync",ifelse(Collection_ID%in% tested,"sync","no_sync")))  #filter(Fungicide_status== "sync")                                                #ifelse(Collection_ID%in% montcalm,"sync", "no_sync")))) %>% 
  #filter(Fungicide_status== "sync")

sum(selection$Field=="lan" ) + sum(selection$Field=="mon" )+sum(selection$Field=="mont" )
bursapehelnchus <- bettle %>% mutate(Selection_status=
                       ifelse(Collection_ID%in% montcalm, "yes", "no"))
                                  
```
 
 
 
#THIRD PART OF DISSERTATION, Characterize the population structure and genetic diversity of _S. sclerotiorum_ populations from soybean throughout the North Central United States and Mexico
#Background
_S. sclerotiorum_ has a haploid somatic phase and its reproduction could be asexual by sclerotia and sexual by self-fertilization or outcrossing (Kohn 1995). Population studies on S. sclerotiorum have revealed a predominantly clonal mode of reproduction (Cubeta et al. 1997; Kohli and Kohn 1998) with some evidence of outcrossing in a few regions (Atallah et al. 2004; Kohli and Kohn 1998). Recent studies identified mostly clonal genotypes from common bean with no significant difference between the most distant places (France, Australia and the U.S.), which is contrary to that expected because S. sclerotiorum is a soil borne pathogen. Results further indicated genotypes from Mexico and those isolated in 2005 were the strongest predictors of the genetic variation. Based on that, more isolates from the U.S. and Mexico are needed to further clarify the significance of genetic variation associated with Mexico. Thus, the aims of my research objective is to use SSR genotyping to estimate levels of genetic diversity within populations of S. sclerotiorum from soybean in the North Central United States and from dry bean in Mexico. Therefore, __the hypothesis is that Mexico is the point of origin of _S. sclerotiorum_ where sexual outcrossing ocurred  possibly due to more conducive environmental conditions?__


Select hierarchieally isolates by space according to genotype 
 
 #Selection of the country of Mexico  
```{r}

wm2_ <-  wm1 %>%
  mutate(Fungicide_status = ifelse(Collection_ID  %in% tested,# Creating a new column based wheater or not fungicides pretested
                                   "tested", "no_tested")) %>%# Creating a new column based wheater or not fungicides pretested
  group_by(Field) %>% mutate (num_isolates_per_field = ifelse(n() > 2, "2>isolates/Field", "<3isolates/Field")) %>%#Creating a new column if the number of isolates/field is major than 2 "<2isolates/Field"
  ungroup() %>%
  filter(Collection_ID  %in% mexican_isolates) #Filtering mexican isolates


wm2_$num_isolates_per_field <-#Converting  as factor the levels of the new columns
  as.factor(wm2_$num_isolates_per_field)




```
###Cleaning and making some arrangemments from the inventory so far, run twice
```{r}
wm2_$long <- as.numeric(as.character(wm2_$long))# Transforming long variables to numeric
wm2_$long <- (wm2_$long) * -1 # Transforming the long variable to the Hemispher West side



```
###Data wrangling
```{r, echo=FALSE}
# creating a subdata of selection_Mexico
selection_Mexico <-
  wm2_ %>% select(
   Collection_ID,
    State,
    County,
    Field,
    Year,
    Origin,
    long,
    lat,
    hyphal_tip..ht.,
    DNA.Extraction,
    Serial_agar_dilution_for_creating_the_model_..N.20.,
    Form.ID,
    Fungicide_status,
    num_isolates_per_field
  ) %>%
  group_by(State, County, Field) %>%# grouping in order to select those with >=3 isolates/field/County/State 
  mutate(num_ISO_per_field = n()) %>%# Creating the column to count them 
  ungroup %>%
  group_by(State, County) %>%# grouping in order to select those   fields(>=3 isolates)/County/State 
  mutate(num_field_County = length(unique(factor(Field)))) %>%# Creating the column to count them 
  select(# Ordering the collumns to easy visulalization
    Collection_ID,
    State,
    County,
    Field,
    num_ISO_per_field,
    num_field_County,
    Year,
    Origin,
    long,
    lat,
    hyphal_tip..ht.,
    DNA.Extraction,
    Serial_agar_dilution_for_creating_the_model_..N.20.,
    Form.ID,
    Fungicide_status,
    num_isolates_per_field
  )


#Subdata for randomly selection of those 3 isolates/field/county/state
selection_Mexico1 <- selection_Mexico %>%
  ungroup() %>%
  group_by(State, County, Field) %>%#grouping in order to select 3 isolates/field/County/State
  filter(num_ISO_per_field > 10) %>% 
         #Fungicide_status == "no_tested") %>%# Filtering those fields with more than 10 isolates/ field and also by no fungicide pre-tested. Here the original idea was to filter by 3 >= isolates but  it does not not work with the next pipe available from tidyverse
  sample_n(3, replace = F)#Randomly sampled 3 isolates from the 10 available from each group

# Subdata another one to work with those groups with less than 10 isolates and more or equal to 3 isolates. The way to do it was to rest to the ones done
selection_Mexico2 <- selection_Mexico %>% # using the dataframe Selection_Mexico 
  ungroup() %>%
  group_by(State, County, Field) %>%#grouping in order to select 3 isolates/field/County/State
  filter(!num_ISO_per_field %in% selection_Mexico1$num_ISO_per_field) %>%
  #filter(Fungicide_status == "no_tested") %>% #filtering those already done 
  ungroup()
```
###Finding randomly either by function or by hand writing  3 samples for each (by sample function) field for those groupings with less than 10 isolates/field. In those cases of error it is due there is less than 3 isolates per field but it still being okay
```{r, echo=FALSE}
bbb <-
  sample(selection_Mexico2$Collection_ID,# sampling 3 for each group
         size = 3,
         replace = FALSE)


```
###Once all filters, add a new column called "selected" to the master mix dataframe     
```{r}
selection_Mexico <- selection_Mexico %>%
  mutate(
    Status = case_when(
      Collection_ID %in% selection_Mexico1$Collection_ID ~ "selected",# filtering and naming "selected"" in case of in
      Collection_ID %in% bbb ~ "selected",
      TRUE ~ "not_selected"# in case of not, naming "not_selected""
    )
  )

#Plot of Mexico selection
eyebrow <- mexico_base + geom_point(
  data = selection_Mexico,
  aes(
    x = long,
    y = lat,
    color = num_isolates_per_field# by color the level of selection
    #shape = Year# by shape the level of Year
  ),
  inherit.aes = FALSE,
  na.rm = TRUE,
  size = 2
) + theme(legend.position = "bottom") + #facet_wrap(~ Fungicide_status) + #Wrapping by the # of isolates/field
  scale_color_manual(values = c("red", "blue")) + theme(legend.position = "bottom") + labs(title = "Sclerotinia sclerotiorum Greenbean from Sinaloa, Mexico") + theme(plot.title = element_text(
  size = 10,
  face = "bold",
  hjust = 0.5,
  family = "Helvetica"
))
eyebrow

```

```{r}
##Summarizing selected by Mexico
selection_Mexico3 <-
  selection_Mexico %>% filter(Status == "selected") %>% ungroup()
```


```{r, warning= FALSE}
# Combining both
selection_FINAL_objective3 <-
  selection_FINAL_objective1 %>% bind_rows(selection_Mexico3) #%>% filter(Fungicide_status == "no_tested")#and also by no fungicide pre-tested
```


```{r, warning= FALSE}


# Combining both
selection_FINAL_objective3_filterring <-
  selection_FINAL_objective3 %>%  
   filter(DNA.Extraction == "No DNA extraction") #by no DNA extraction
```

```{r}
#Nebraska
WM1S <- selection %>%# creating a subdata of NE
  group_by(State, Origin) %>%
  filter(Origin == "Farmer fields", State == "NE") %>%
  ungroup()

WM1S$DNA.Extraction = factor(WM1S$DNA.Extraction ,
                             levels = c("No DNA extraction", "Yes DNA extraction"))#Ordering the levels of DNA.Extraction
NE_county <- subset(counties, region == "nebraska")# Subsetting just for NE state

NE_county_base <-# Creating the base to use for next plots
  ggplot(data = NE_county,
         mapping = aes(x = long, y = lat, group = group)) +
  coord_fixed(1.3) +
  geom_polygon(color = "black", fill = "grey")

#Plot of NE selection
water <- NE_county_base + geom_point(
  data = WM1S,
  aes(
    x = long,
    y = lat,
    #color = Status, #by color the level of selection
    #shape = Year,# by shape the level of Year
    color = DNA.Extraction# by size the level of DNA.Extraction
  ),
  inherit.aes = FALSE,
  na.rm = TRUE
) + theme(legend.position = "bottom") + facet_wrap( ~ num_isolates_per_field) + #Wrapping by the # of isolates/field
  scale_color_manual(values = c("red", "blue")) + theme(legend.position = "bottom") + labs(title = "Sclerotinia sclerotiorum Soybean from Nebraska") + theme(plot.title = element_text(
    size = 10,
    face = "bold",
    hjust = 0.5,
    family = "Helvetica"
  ))
water




```
 
 
 
```{r}
#install.packages("geosphere")
#library(geosphere)

combiningfungicides_outliersEC50DC <- read.csv("data/combiningfungicides_outliersEC50DC.csv")
combiningfungicides_outliersEC50DC <- combiningfungicides_outliersEC50DC %>% select(-X)
combiningfungicides_outliersEC50DC$ID <- as.character(combiningfungicides_outliersEC50DC$ID)
combiningfungicides_outliersEC50DC$ID[2]="2220"#these the ID of the H-01

  
  
 wm1_with_isolateshighEC50 <- wm1 %>%  mutate(field_selection = ifelse(Collection_ID  %in% combiningfungicides_outliersEC50DC$ID,
                                   "field_selected", "field_no_selected")) %>% filter(field_selection =="field_selected")

  
#
dat_long_lat <- selection_FINAL_objective1 %>% 
  select(State, County, Field, long, lat) %>% 
  unique()
  
#install.packages("sf")
#install.packages("lwgeom")
# library(sf)
# library(lwgeom)

# Create example data frame
#df <- data.frame(latitude = c(49.48609,-8.14671,11.28625),
#                 longitude = c(8.463678,143.05793,-11.18285))

# COnvert to sf object
long_lat_sf <- st_as_sf(dat_long_lat, coords = c("long", "lat"))

# Set the projection as ESPG 4326 (long_lat)
st_crs(long_lat_sf) <- 4326

# Apply the st_distance function
dist_m <- st_distance(long_lat_sf)

# Combine with df
df_long_lat <- dat_long_lat %>%
  mutate(`distance-latlon1` = as.numeric(dist_m[, 1]), 
         `distance-latlon2` = as.numeric(dist_m[, 2]),
         `distance-latlon3` = as.numeric(dist_m[, 3])) 

# Replace 0 with NA
df_long_lat[df_long_lat == 0] <- NA

```
 
```{r}
```
 
 